pkg/memo/memo.go

```diff
@@ -51,7 +51,7 @@ func (m *InboundMemo) EncodeToBytes() ([]byte, error) {
 
 // DecodeFromBytes decodes a InboundMemo struct from raw bytes
 //
-// Returns an error if given data is not a valid memo
+// Returns nil memo if given data can't be decoded as a memo.
```

Comment:
If this has to be processed differently we could also reutrn a parsing error to be handled by the calling function, not sure returning an inboundMemo as a pointer makes the API simpler

Reply:
We'll have to support both legacy memo and standard memo at some point. Zetaclient has to try deserializing given bytes as a standard memo struct. If serialization fails, zetaclient knows that the given bytes is not a standard memo and continue processing the bytes as legacy memo.

**The problem:**
       If the given bytes is well formatted in the standard way but still carry improper values (e.g., invalid address), we SHOULD NOT continue and process it as a legacy memo. So we kinda need an extra information to tell: `are these bytes a standard memo or not ?`.
       
**The way to solve:**
      Before phasing out legacy memo. Let the API `DecodeFromBytes` to return a hint (nil or not nil) to tell `whether the data is a standard memo or not`, otherwise we'll need a separate API `IsStandardMemo(memoBytes []byte) bool` to do so.
     
   After phasing out legacy, let the API `DecodeFromBytes` to return `nil` on error.

Reply:
To make the logic clear, here is the refactor: https://github.com/zeta-chain/node/pull/3025/commits/b2c149c4124f27c83f2ef6f4a55f76f8c3534433